src/context.rs

```diff
@@ -30,19 +31,83 @@ impl<'a> QueryMatchContext<'a> {
         file_contents: &'a [u8],
         rule: &'a ResolvedRule,
         reported_any_violations: &'a AtomicBool,
+        context: &'a Context,
     ) -> Self {
         Self {
             path,
             file_contents,
             rule,
             reported_any_violations,
+            context,
         }
     }
 
     pub fn report(&self, violation: Violation) {
         self.reported_any_violations.store(true, Ordering::Relaxed);
         print_violation(&violation, self);
     }
+
+    pub fn get_node_text(&self, node: Node) -> &str {
```

Comment:
I kind of like this API feel of exposing a bunch of "convenience" methods on the `QueryMatchContext`

But it's a little arbitrary what would go on here or not